src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2294,7 +2294,9 @@ public override IReadOnlyList<SyntaxNode> GetParameters(SyntaxNode declaration)
             var list = GetParameterList(declaration);
             return list != null
                 ? list.Parameters
-                : SpecializedCollections.EmptyReadOnlyList<SyntaxNode>();
+                : declaration is SimpleLambdaExpressionSyntax simpleLambda
+                    ? new[] { simpleLambda.Parameter }
+                    : SpecializedCollections.EmptyReadOnlyList<SyntaxNode>();
```

Comment:
this helper method just reutrns a list of ParameterNodes.  In this case, it's fine to return the single-parameter as the only item of that list.   Note that SimpleLambdaExpressionSyntax.Parameter is typed as ParameterNode, not Identifier, so higher lever features can work uniformly over parameter nodes.